src/lit-html.ts

```diff
@@ -12,22 +12,30 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
+/**
+ * TypeScript has a problem with precompiling templates literals
+ * https://github.com/Microsoft/TypeScript/issues/17956
+ */
+const cacheTemplates = ((t: any) => t() === t())(() => ((s: TemplateStringsArray) => s)``);
```

Comment:
I think given that this is used backwards in the ternary below (and I did it too!), that it should have a clear comment and clearer name. I originally suggested `cachesTemplates` as in "the browser caches templates like it should", but that one extra "s" may be too subtle. Maybe `templateStringsAreCached`, so that the use reads like:

```ts
const key = templateStringsAreCached ? strings : strings.join('{{typescriptProblems}}');
```